news/_posts/2019-15-01-resilient-php-applications.md

```diff
@@ -0,0 +1,108 @@
+---
+layout: post
+title:  "Resilient PHP applications"
+subtitle: "Introducing the PrestaShop Circuit Breaker library"
+date:   2019-01-10 10:00:00
+authors: [ mandrieu ]
+icon: icon-blaster
+tags:
+ - 1.7
+---
+
+Using an extensible e-commerce platform like PrestaShop is awesome for both developers and e-merchants. Relying on an open source project with the possibility to add themes and plugins available in the store allows every business to create a shop that fits their needs.
+
+But, no matter how well designed is your project and how many tests you have, at some point your website can become unavailable.
+
+This can be because of your database, or calls to third party services (payment methods API, PrestaShop REST API to retrieve modules information) are unavailable.
+
+Instead of thinking everything will be ok, we need - as developers - to manage the failures of our systems.
+
+We need to create resilient systems, to create websites able to recover quickly from network disasters.
+
+## The risks of depending to external services
+
+Assuming all external services will works fine is risky because you don't think
```

Comment:
```suggestion
Assuming all external services will works fine is risky because you don't think about what should happens in case of unavailability:
```